packages/workerd/src/plugin.ts

```diff
@@ -84,13 +85,29 @@ export async function createWorkerdDevEnvironment(
   config: ResolvedConfig,
   pluginOptions: WorkerdEnvironmentOptions,
 ) {
+  const workerPath = fileURLToPath(new URL("./worker.js", import.meta.url));
+  const workerContent = readFileSync(workerPath, "utf-8");
+  const viteModuleRunnerPath = fileURLToPath(
+    import.meta.resolve("vite/module-runner"),
+  );
+  const viteModuleRunnerContent = readFileSync(
+    viteModuleRunnerPath,
+    "utf-8",
+    // avoid new AsyncFunction during import side effect
+  ).replace(`new AsyncFunction("a", "b", body).toString()`, `""`);
```

Comment:
Otherwise we get:

```
service core:user:: Uncaught Error: EvalError: Code generation from strings disallowed for this context
  at vite/module-runner:21:37
  at vite/module-runner:24:3
error when starting dev server:
MiniflareCoreError [ERR_RUNTIME_FAILURE]: The Workers runtime failed to start. There is likely additional logging output above.
    at #assembleAndUpdateConfig (/home/hiroshi/code/personal/vite-environment-examples/node_modules/.pnpm/miniflare@3.20240925.0/node_modules/miniflare/dist/src/index.js:9828:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Mutex.runWith (/home/hiroshi/code/personal/vite-environment-examples/node_modules/.pnpm/miniflare@3.20240925.0/node_modules/miniflare/dist/src/index.js:3627:16)
    at async #waitForReady (/home/hiroshi/code/personal/vite-environment-examples/node_modules/.pnpm/miniflare@3.20240925.0/node_modules/miniflare/dist/src/index.js:9885:5)
    at async Miniflare2._getProxyClient (/home/hiroshi/code/personal/vite-environment-examples/node_modules/.pnpm/miniflare@3.20240925.0/node_modules/miniflare/dist/src/index.js:10009:5)
    at async #getProxy (/home/hiroshi/code/personal/vite-environment-examples/node_modules/.pnpm/miniflare@3.20240925.0/node_modules/miniflare/dist/src/index.js:10069:25)
    at async createWorkerdDevEnvironment (file:///home/hiroshi/code/personal/vite-environment-examples/packages/workerd/dist/index.js:190:14)
    at async _createServer (file:///home/hiroshi/code/personal/vite-environment-examples/node_modules/.pnpm/vite@https+++pkg.pr.new+vite@70bb8de_@types+node@20.14.11_terser@5.31.3/node_modules/vite/dist/node/chunks/dep-DsfahqHZ.js:51979:26)
    at async CAC.<anonymous> (file:///home/hiroshi/code/personal/vite-environment-examples/node_modules/.pnpm/vite@https+++pkg.pr.new+vite@70bb8de_@types+node@20.14.11_terser@5.31.3/node_modules/vite/dist/node/cli.js:745:20)
```